app/src/main/java/org/fossasia/openevent/app/main/MainActivity.java

```diff
@@ -114,7 +115,7 @@ public void onDrawerClosed(View drawerView) {
 
                 if (id == R.id.nav_logout)
                     showLogoutDialog();
-                else if (eventId != -1)
+                else if (eventId != -1 || id == R.id.nav_settings)
```

Comment:
yes, you have added it to the `mainMenu` group so it will be always shown. Only visibility condition is applied to the `subMenu` which contains event specific items.